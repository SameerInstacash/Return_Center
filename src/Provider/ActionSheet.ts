import { Injectable } from "@angular/core";
import { ActionSheetController } from "@ionic/angular";
import { callbackify } from "util";

@Injectable()
export class ActionSheetProvider {

    public LangDict : any = {};

    constructor(private actionSheetCtrl: ActionSheetController) {

        let retrievedObject = localStorage.getItem("languageJSON");
        let json = JSON.parse(retrievedObject)
        this.LangDict = JSON.parse(json)

    }

    async presentActionSheet() {
        
        const actionSheet = this.actionSheetCtrl.create({
            
            buttons : [{
                text : this.LangDict['Check_Device'],
                role : '1',
               
            },
            {   
                text : this.LangDict['Product_Info'],
                role : '2',
            
            }]
        });

        (await actionSheet).present();
        //const { role, data } = await (await actionSheet).onDidDismiss();

        let data = await (await actionSheet).onDidDismiss();
        return data

    }


    async presentActionSheetForDispatch() {
        
        const actionSheet = this.actionSheetCtrl.create({
            
            buttons : [{
                text : this.LangDict['Dispatch'],
                role : '1',
               
            },
            {   
                text : this.LangDict['Product_Info'],
                role : '2',
            
            }]
        });

        (await actionSheet).present();
        //const { role, data } = await (await actionSheet).onDidDismiss();

        let data = await (await actionSheet).onDidDismiss();
        return data

    }


    async presentActionSheetForProductInfo() {
        
        const actionSheet = this.actionSheetCtrl.create({
            
            buttons : [{
                text : this.LangDict['Product_Info'],
                role : '1',
               
            },
            {   
                text : this.LangDict['Dismiss'],
                role : '2',
            
            }]
        });

        (await actionSheet).present();
        //const { role, data } = await (await actionSheet).onDidDismiss();

        let data = await (await actionSheet).onDidDismiss();
        return data

    }


}