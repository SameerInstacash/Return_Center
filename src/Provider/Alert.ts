import { Injectable } from "@angular/core";
import { AlertController } from "@ionic/angular";

@Injectable()
export class AlertProvider {

    public LangDict : any = {};
    alert: any = null;

    constructor(private alertCtrl: AlertController) {

        let retrievedObject = localStorage.getItem("languageJSON");
        let json = JSON.parse(retrievedObject)
        this.LangDict = JSON.parse(json)

    }

    // Global Alert
    async presentAlertOnView(title:String,message:String) {
        const alert = this.alertCtrl.create({
            header : title as string,
            subHeader : message as string,
            buttons : [this.LangDict['Ok']]
        });
        (await alert).present();
    }

    async presentLogoutAlert(title,message,callBack) {

        const alert = this.alertCtrl.create({
            header : title as string,
            subHeader : message as string,

            buttons : [{
                text: this.LangDict['No'],
                handler: (alertData) => {
                    callBack(true);
                }
            },{
                text: this.LangDict['Yes'],
                handler: (alertData) => {
                    callBack(false);
                }
            }]
        });
        (await alert).present();
    }

    // Language Alert
    async presentLanguageCountryAlert(langs, callback) {
        
        if (this.alert == null) {
            
            const alert = this.alertCtrl.create({
                cssClass: 'langCountry',
                header: this.LangDict['Alert'],
                inputs: langs,
                buttons: [
                    {
                        text: this.LangDict['Cancel'],
                        role: 'cancel',
                        cssClass: 'secondary',
                        handler: (alertData) => {
                            callback('Hi');
                        }
                    },
                    {
                        text: this.LangDict['Ok'],
                        handler: (alertData) => {
                            callback(alertData);
                        }
                    }
                ]
            });
            (await alert).present();
        }
    }



}