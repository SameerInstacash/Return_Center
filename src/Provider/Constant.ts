
export const AppUserName = 'ICReturnCenter';
export const AppApiKey = '7a8f1929a92334bb1555d25293919293';


// ALL APIS
//export const baseUrl = 'https://getinstacash.in/zwReturnCenter/api/v1/public/';
export const loginApi = 'login';
export const getNumberOfItemApi = 'getNumberOfItem';
export const getStockApi = 'getStock';
export const productVerificationModifierApi = 'productVerificationModifier';
export const verifyProductApi = 'verifyProduct';
export const getLocationApi = 'getLocation';
export const getStatusApi = 'getStatus';
export const addToStockApi = 'addToStock';
export const changeLocationApi = 'changeLocation';
export const advanceSearchApi = 'advanceSearch';


//"cordova-plugin-nativegeocoder": {
//"LOCATION_WHEN_IN_USE_DESCRIPTION": "Use geocoder service"
//},


export const SEARCH_TYPE_ORDER_ID = "orderId";
export const SEARCH_TYPE_IMEI = "IMEI";
export const SEARCH_TYPE_DATE = "date";
export const SEARCH_TYPE_FROM_DATE = "fromDate";
export const SEARCH_TYPE_TO_DATE = "toDate";
export const SEARCH_TYPE_PAGE = "page";
