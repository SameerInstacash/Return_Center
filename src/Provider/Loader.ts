import { LoadingController } from "@ionic/angular";
import { Injectable } from "@angular/core";

@Injectable()
export class LoaderProvider {

    public LangDict : any = {};

    constructor(public loader:LoadingController) {

        let retrievedObject = localStorage.getItem("languageJSON");
        let json = JSON.parse(retrievedObject)
        this.LangDict = JSON.parse(json)

    }

    async showLoader(strMessage: string) {
        const loading = this.loader.create({
            message: strMessage,
            //message: '<ion-img src="/assets/img/india.png" alt="loading..."></ion-img>',
            cssClass: 'scale-down-center',
            translucent: true,
            showBackdrop: true,
            spinner: null,
            duration: 4000
        });

        (await loading).present();
    }

    dismissLoader() {
        this.loader.dismiss();
    }

}