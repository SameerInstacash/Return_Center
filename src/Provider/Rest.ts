import { Injectable } from "@angular/core";
import { HttpClient } from '@angular/common/http';
import { HTTP } from '@ionic-native/http/ngx';
import { Network } from '@ionic-native/network/ngx';


@Injectable()
export class RestProvider {

    public LangDict : any = {};

    constructor(private network: Network, 
        private http: HTTP,
        private httpClient: HttpClient) {

          let retrievedObject = localStorage.getItem("languageJSON");
          let json = JSON.parse(retrievedObject)
          this.LangDict = JSON.parse(json)

    }

    // Get Request
    GetRequest(apiName: string) {
        return new Promise((resolve, reject) => {
            let baseUrl = localStorage.getItem('base_url');

            if (this.network.type == 'none') {
                
                reject(this.LangDict['Please_Check_Your_Network_Connection']);
                
            }else {
                
                let headers = {
                    'Content-Type': 'text/xml; charset=utf-8',
                    'Access-Control-Allow-Methods': 'GET',
                    'Access-Control-Allow-Origin': '*',
                    'Access-Control-Allow-Headers': "Access-Control-Allow-Headers, Access-Control-Allow-Origin, Access-Control-Request-Method"
                };

                this.httpClient.get(baseUrl + apiName).subscribe((response) => {
                    debugger
                    console.log(response);
                    resolve(response);
                });

            }
        }); 
    }

    // POST Request
    postRequest(apiName : string, parameters) {

        return new Promise((resolve, reject) => {
            let baseUrl = localStorage.getItem('base_url');
            //let baseUrl = 'https://getinstacash.in/zwReturnCenter/api/v1/public/';

            //let IN = 'https://getinstacash.in/instaCash/api/v5/public/';
            //let MY = "https://getinstacash.com.my/instaCash/api/v5/public/"
            //let SG = "https://getinstacash.com.sg/instaCash/api/v5/public/"
            //let TW = "https://getinstacash.com.tw/instaCash/api/v5/public/"
            //let PH = "https://ph.getinstacash.com/instaCash/api/v5/public/"
            //let TH = "https://th.getinstacash.com/instaCash/api/v5/public/"

            
            if (this.network.type == 'none') {
                reject(this.LangDict["Please_Check_Your_Network_Connection"]);
            }else {

                let headers = {
                    'Accept':'*/*',
                    'Content-Type':'application/x-www-form-urlencoded'
                };
                

                /*
                let body = new URLSearchParams();
                body = parameters
                this.httpClient.post((baseUrl + apiName), body, { headers : headers }).subscribe((response) => {
                    debugger
                    console.log(response);
                    resolve(response);
                });
                */

            
                this.http.setDataSerializer('urlencoded');
                this.http.post((baseUrl + apiName), parameters, headers).then((response) => {
                    
                    console.log(response);
                    resolve(response);
                }).catch((error) => {
                    
                    console.log("error==>"+JSON.stringify(error));

                    reject(this.LangDict["Something_Went_Wrong_Please_Try_Again_Later."]);
                });
                
            }
        }); 
    }
 

    // GET
    languageDownload(type: string) {
        return new Promise((resolve, reject) => {

          //let baseUrl = localStorage.getItem('base_url');

            if (this.network.type!='none') {

              let headers = {
                //'Content-Type': 'text/xml; charset=utf-8',
                //'Access-Control-Allow-Methods': 'GET',
                'Access-Control-Allow-Origin': '*',
                //'Access-Control-Allow-Headers': "Access-Control-Allow-Headers, Access-Control-Allow-Origin, Access-Control-Request-Method"
              };
             
              //this.http.get(baseUrl + type, {}, headers)
              this.http.get(type, {}, headers)
                .then(data => {
                  
                  console.log(data["data"]);
                  resolve(data);
                })
                .catch(error => {
                  
                  reject(this.LangDict["Something_Went_Wrong_Please_Try_Again_Later."]);
                });
            }else{
                
                reject(this.LangDict["Please_Check_Your_Network_Connection"]);
            }
        });
    }

    postApplicationJsonData(apiName,data) {
        return new Promise((resolve, reject) => {

        let baseUrl = localStorage.getItem('base_url');
        //let baseUrl = "https://getinstacash.in/zwReturnCenter/api/v1/public/";

        if(this.network.type!='none'){

          let headers = {
            'Accept':'application/json',
            'Content-Type':'application/json'
          };
       
          this.http.setDataSerializer('json');
          this.http.post(baseUrl+apiName, data, headers)
            .then(data => {
              //console.log(data["data"]);
              resolve(data);
            })
            .catch(error => {
              reject(error);
              console.log("error==>"+JSON.stringify(error));
            });
        }else{
            reject(this.LangDict["Please_Check_Your_Network_Connection"]);        
        }
    });
    }
    
    //*
    post(data) {
        return new Promise((resolve, reject) => {

          let baseUrl = localStorage.getItem('base_url');
          
          if(this.network.type!='none'){
          let headers = {
            'Accept':'application/json',
            'Content-Type':'application/json'
          };
       
          this.http.setDataSerializer('json');
          this.http.post(baseUrl+'advanceSearch', data, headers)
          //this.http.post("https://getinstacash.in/zwReturnCenter/api/v1/public/advanceSearch", data, headers)
            .then(data => {
              console.log(data["data"]);
              resolve(data);
            })
            .catch(error => {
              reject(error);
            });
          }else{
            reject(this.LangDict["Please_Check_Your_Network_Connection"]);        
          }
        });
      }
      //*/


}
