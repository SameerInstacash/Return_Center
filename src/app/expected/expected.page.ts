import { Component, OnInit } from '@angular/core';
import { ActivatedRoute, NavigationExtras, Router } from '@angular/router';
import { AppApiKey, AppUserName, changeLocationApi, getNumberOfItemApi, getStockApi } from 'src/Provider/Constant';
import { RestProvider } from 'src/Provider/Rest';
import { AlertProvider } from 'src/Provider/Alert';
import { LoaderProvider } from 'src/Provider/Loader';
import { ActionSheetProvider } from 'src/Provider/ActionSheet';


@Component({
  selector: 'app-expected',
  templateUrl: './expected.page.html',
  styleUrls: ['./expected.page.scss'],
})
export class ExpectedPage implements OnInit {

  public LangDict : any = {};

  public arrFilteredStock : any[];
  public arrStock : any[];
  public selectedLocationId : any;
  public strCurrentLocation : string;
  public apiFlag : any;
  navCtrl: any;
  public isAdvanceSearch = false

  constructor(private route: ActivatedRoute, private router: Router, 
    private alert: AlertProvider, private loader: LoaderProvider,
    private rest: RestProvider, private actionSheet: ActionSheetProvider) { 

    this.route.queryParams.subscribe((params) => {
      if (this.router.getCurrentNavigation().extras.state) {

        this.apiFlag = this.router.getCurrentNavigation().extras.state.flag;
        this.isAdvanceSearch = this.router.getCurrentNavigation().extras.state.isSearch;

        if (this.apiFlag == 'apiTrue') {

          if (this.isAdvanceSearch == false) {
            
            this.selectedLocationId = this.router.getCurrentNavigation().extras.state.user;
            this.strCurrentLocation = this.router.getCurrentNavigation().extras.state.pageTitle;
  
            this.getStockItemsOnStockPage();
          }else {
            // Load hold data
            
          }

        }else {
          
          this.isAdvanceSearch = true
          this.arrStock = this.router.getCurrentNavigation().extras.state.searchData;
        }
        
      }
    });

    
    let retrievedObject = localStorage.getItem("languageJSON");
    let json = JSON.parse(retrievedObject)
    this.LangDict = JSON.parse(json)

  }

  ngOnInit() {

  }

  search(query) {
    if (!query) { // revert back to the original array if no query
      this.arrStock = this.arrFilteredStock;
    } else { // filter array by query

      /*
      this.arrStock = this.arrFilteredStock.filter((user) => {
        return (user.productName.toLowerCase.includes(query.toLowerCase) || 
                user.productIMEI.toLowerCase.includes(query.toLowerCase) || 
                user.productOrderId.toLowerCase.includes(query.toLowerCase)
               );
      })*/

      this.arrStock = this.arrFilteredStock.filter((user) => {
        return user.productName.toLowerCase().indexOf(query.toLowerCase()) > -1;
      });

    }
  }


  getStockItemsOnStockPage() {

    let postData = {
      "userName" : AppUserName,
      "apiKey" : AppApiKey,
      "returnCenterId" : localStorage.getItem("returnCenterId"),
      "locationId" : this.selectedLocationId,
    }
        
    //console.log(postData);
    this.loader.showLoader(this.LangDict['Loading...']);

    this.rest.postRequest(getStockApi,postData).then((responseData) => {

      this.loader.dismissLoader();

      //console.log(JSON.parse(responseData['data']));
      //console.log("true==>"+JSON.stringify(JSON.parse(responseData['data'])));

      if (JSON.parse(responseData['data']).status == 'Success') {
        
        //console.log(JSON.parse(responseData['data']));
        this.arrStock = JSON.parse(responseData['data']).msg;
        this.arrFilteredStock = this.arrStock

      }else {
        debugger
        this.alert.presentAlertOnView(this.LangDict['Error'],JSON.parse(responseData['data']).msg);
      }
    })
    .catch((error) => {
      debugger
      this.alert.presentAlertOnView(this.LangDict['Error'],error);
    });

  }


  getDetailOfSelectedStockItem(index: any) {

    if (this.apiFlag != 'apiTrue') {
      this.selectedLocationId = this.arrStock[index].currentLocation;
      //this.strCurrentLocation = this.arrStock[index].
    }


    if (this.selectedLocationId == '1') {

      this.actionSheet.presentActionSheet().then((callBackData) => {
        if (callBackData.role == '1') {
  
          let navigationExtras: NavigationExtras = {
            state: {
              currentDevice : this.arrStock[index],
              currentLocation : this.strCurrentLocation,
              locationID : this.selectedLocationId,
              isSearch : this.isAdvanceSearch
            }
          };
          
          this.router.navigate(['device-check-up'], navigationExtras);
  
        }else if (callBackData.role == '2') {
  
          let navigationExtras: NavigationExtras = {
            state: {
              currentDevice : this.arrStock[index],
              currentLocation : this.strCurrentLocation,
              locationID : this.selectedLocationId,
              isSearch : this.isAdvanceSearch
            }
          };
          
          this.router.navigate(['product-info'], navigationExtras);
  
        }else {
           // Just dismiss
        }
      });

    }else if (this.selectedLocationId == '4') {

      this.actionSheet.presentActionSheetForDispatch().then((callBackData) => {
        if (callBackData.role == '1') {

          // Call changeLocation Api for move item from stock to dispatch
          this.callChangeLocationApi(index);
          
        }else if (callBackData.role == '2') {
          
          let navigationExtras: NavigationExtras = {
            state: {
              currentDevice : this.arrStock[index],
              currentLocation : this.strCurrentLocation,
              locationID : this.selectedLocationId
            }
          };
          
          this.router.navigate(['product-info'], navigationExtras);
  
        }else {
           // Just dismiss
        }
      });

    }else {

      // Just show product info on next page
      this.actionSheet.presentActionSheetForProductInfo().then((callBackData) => {
        if (callBackData.role == '1') {

          let navigationExtras: NavigationExtras = {
            state: {
              currentDevice : this.arrStock[index],
              currentLocation : this.strCurrentLocation,
              locationID : this.selectedLocationId
            }
          };
          
          this.router.navigate(['product-info'], navigationExtras);
          
        }else if (callBackData.role == '2') {
          
          // Just dismiss
  
        }else {
           // Just dismiss
        }
      });

      
    }

  }; 


  callChangeLocationApi(selIndex) {

    let postData = {
      "userName" : AppUserName,
      "apiKey" : AppApiKey,
      "returnCenterId" : localStorage.getItem("returnCenterId"),
      "id" : this.arrStock[selIndex].id,
      "currentLocation" : '4',
      "newLocation" : '8'
    }
  
    //console.log(postData);
    this.loader.showLoader(this.LangDict['Loading...']);
  
    this.rest.postRequest(changeLocationApi,postData).then((responseData) => {
      
      this.loader.dismissLoader();
      
      //console.log(JSON.parse(responseData['data']));
      //console.log("true==>"+JSON.stringify(JSON.parse(responseData['data'])));
  
      if (JSON.parse(responseData['data']).status == 'Success') {
        
        let resultData = JSON.parse(responseData['data']).msg;
        console.log(resultData);
  
        this.getStockItemsOnStockPage()      
  
      }else {
        debugger
        this.alert.presentAlertOnView(this.LangDict['Error'],JSON.parse(responseData['data']).msg);
      }
    })
    .catch((error) => {
      debugger
      this.alert.presentAlertOnView(this.LangDict['Error'],error);
    });
  
  }
  

}


