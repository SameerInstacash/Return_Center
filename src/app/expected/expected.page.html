<ion-header>
  <ion-toolbar>
    <ion-buttons slot="start">
      <ion-back-button defaultHref="" class="gray1"></ion-back-button>
    </ion-buttons>
    <ion-title class="pL80">{{strCurrentLocation}}</ion-title>
  </ion-toolbar>

  <ion-searchbar (ionChange)="search($event.target.value)"></ion-searchbar>
  
</ion-header>

<ion-content class="expected">

  <!-- List of Text Items -->
<ion-list>

  <!-- <div>
  <ion-item *ngFor="let item of arrStock; let i = index" (click)='getDetailOfSelectedStockItem(i)'>
    <ion-label>{{item.productName}}</ion-label>
    <ion-label slot="end">{{item.productPrice}}</ion-label>
  </ion-item>
  </div> -->

  <ion-card *ngFor="let item of arrStock; let i = index" (click)='getDetailOfSelectedStockItem(i)'>
    <div class="cardPhoneDetail">
      <ion-card-header>
        <ion-card-title>{{item.productName}}</ion-card-title>
        <ion-card-subtitle>{{item.productIMEI}}</ion-card-subtitle>
      </ion-card-header>

      <ion-card-content>
        <p>{{item.productOrderId}}</p>
        <p class="price"><i class="fa fa-rupee"></i> {{item.productPrice}}</p>
      </ion-card-content>
    </div>
    <div class="cardPhone">
      <!-- <ion-img src="../assets/img/mobiles/2.png"></ion-img> -->
      <ion-img [src]="item.productImage"></ion-img>
    </div>
  </ion-card>


</ion-list>

  
  

  <!-- <ion-card>
    <div class="cardPhoneDetail">
      <ion-card-header>
        <ion-card-title>Samsung Galaxy S20 Ultra 5G</ion-card-title>
        <ion-card-subtitle>111238113042853</ion-card-subtitle>
      </ion-card-header>

      <ion-card-content>
        <p>ZWJA294991</p>
        <p class="price"><i class="fa fa-rupee"></i> 1,320</p>
      </ion-card-content>
    </div>
    <div class="cardPhone">
      <ion-img src="../assets/img/mobiles/1.jpg"></ion-img>
    </div>
  </ion-card>

  <ion-card>
    <div class="cardPhoneDetail">
      <ion-card-header>
        <ion-card-title>Samsung Galaxy Grand Neo Plus</ion-card-title>
        <ion-card-subtitle>111238113042853</ion-card-subtitle>
      </ion-card-header>

      <ion-card-content>
        <p>ZWJA294991</p>
        <p class="price"><i class="fa fa-rupee"></i> 2,120</p>
      </ion-card-content>
    </div>
    <div class="cardPhone">
      <ion-img src="../assets/img/mobiles/2.png"></ion-img>
    </div>
  </ion-card>

  <ion-card>
    <div class="cardPhoneDetail">
      <ion-card-header>
        <ion-card-title>Apple iPhone 12 mini</ion-card-title>
        <ion-card-subtitle>111238113042853</ion-card-subtitle>
      </ion-card-header>

      <ion-card-content>
        <p>ZWJA294991</p>
        <p class="price"><i class="fa fa-rupee"></i> 12,120</p>
      </ion-card-content>
    </div>
    <div class="cardPhone">
      <ion-img src="../assets/img/mobiles/3.jpg"></ion-img>
    </div>
  </ion-card>

  <ion-card>
    <div class="cardPhoneDetail">
      <ion-card-header>
        <ion-card-title>Apple iPhone 12</ion-card-title>
        <ion-card-subtitle>111238113042853</ion-card-subtitle>
      </ion-card-header>

      <ion-card-content>
        <p>ZWJA294991</p>
        <p class="price"><i class="fa fa-rupee"></i> 12,120</p>
      </ion-card-content>
    </div>
    <div class="cardPhone">
      <ion-img src="../assets/img/mobiles/4.jpeg"></ion-img>
    </div>
  </ion-card>

  <ion-card>
    <div class="cardPhoneDetail">
      <ion-card-header>
        <ion-card-title>Apple iPhone 5s</ion-card-title>
        <ion-card-subtitle>111238113042853</ion-card-subtitle>
      </ion-card-header>

      <ion-card-content>
        <p>ZWJA294991</p>
        <p class="price"><i class="fa fa-rupee"></i> 12,120</p>
      </ion-card-content>
    </div>
    <div class="cardPhone">
      <ion-img src="../assets/img/mobiles/5.jpg"></ion-img>
    </div>
  </ion-card> -->

  
</ion-content>